expyriment_app/main.py

```diff
@@ -1,25 +1,66 @@
 import sys
 import os
 import glob
+from importlib import import_module
 
 import pygame
 import expyriment
 import android
 
+def find_keyword_files(folder):
+    """find all files with keywords in folder and subfolders
+
+    Returns dict of files
+    """
+
+    keywords = ["expyriment", "initialize("]
+    rtn = {}
+    for entry in glob.glob(folder+"/*"):
+        if os.path.isdir(entry):
+            rtn.update(find_keyword_files(entry))
+        elif entry.endswith(".py") and not \
+                                os.path.split(entry)[1].startswith("_"):
+            try:
+                with open(entry) as fl:
+                    lines = fl.readlines()
+            except:
+                lines = ""
+            all_keywords_found = True
+            for key in keywords:
+                if not check_keyword(lines, key):
+                    all_keywords_found = False
+                    break
+            if all_keywords_found:
+                rtn[os.path.split(entry)[-1][:-3]] = entry
+    return rtn
+
+def check_keyword(lines, keyword):
+    """check if keyword occurs in text lines"""
+    for l in lines:
+        if l.find(keyword)>=0:
+            return True
+    return False
+
+def launch_experiment(script, name):
+        expyriment.stimuli.TextScreen(heading="Starting {0}".format(name),
+                        "").present()
+        expyriment.misc.Clock().wait(1000)
+        expyriment.control.defaults.event_logging = 1
+        expyriment.control.defaults.initialize_delay = 0
+        script = os.path.abspath(script)
+        path, pyfile = os.path.split(script)
+        os.chdir(path)
+        sys.argv[0] = script
+        sys.path.insert(1, path)
+        import_module(os.path.splitext(pyfile)[0])
 
 def main():
     android.init()
 
     projects = {}
     for folder in glob.glob("/mnt/*"):
         if os.path.isdir(folder):
-            for project in glob.glob(folder + "/expyriment/*"):
-                if os.path.isdir(project):
-                    path = project + "/*.py"
-                else:
-                    path = folder + "expyriment/*.py"
-                for pyfile in glob.glob(path):
-                    projects[os.path.split(pyfile)[-1]] = pyfile
+            projects.update(find_keyword_files(folder + "/expyriment")
```

Comment:
closing bracket missing
```suggestion
            projects.update(find_keyword_files(folder + "/expyriment"))
```